### DISCUSSION TITLE
Why not use keyv to implement db() 💡

### DISCUSSION THREAD

**Discussion by alkene5** [johnlindquist/kit#1282] [Ideas]
<img width="821" alt="image" src="https://github.com/johnlindquist/kit/assets/61105068/980115c8-1883-473f-b59a-7e745a689f07">

```js
let disk = await store('my-todos')
let todoList = await disk.get('todoList') || []
let writeDisk = async () => await disk.set('todoList', todoList)
```

I am now using store to call keyv data, I think why not keep using db and just change the db implementation to keyv?

**Comment by johnlindquist** [comment]
Great question:

The resulting data of "keyv" includes much more metadata about when properties are written, the the data might be conflict if we swap it out.

kevy also automatically writes to the json file when you call "set", so it may causes some unexpected behavior when using multiple properties vs only writing when calling "write"

I'll keep thinking about it, but I'm still leaning towards no

**Reply by alkene5** [reply]
👌🏻
